overlord/snapstate/snapmgr.go

```diff
@@ -660,17 +645,21 @@ func snapSetupAndState(t *state.Task) (*SnapSetup, *SnapState, error) {
 
 func (m *SnapManager) undoMountSnap(t *state.Task, _ *tomb.Tomb) error {
 	t.State().Lock()
-	ss, snapst, err := snapSetupAndState(t)
+	ss, _, err := snapSetupAndState(t)
 	t.State().Unlock()
 	if err != nil {
 		return err
 	}
 
-	pb := &TaskProgressAdapter{task: t}
-	typ, err := snapst.Type()
+	t.State().Lock()
+	var typ snap.Type
+	err = t.Get("snap-type", &typ)
+	t.State().Unlock()
 	if err != nil {
```

Comment:
for backward compatibility I wonder if shouldn't just ignore ErrNoState here and set typ to app in that case